src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1162,14 +1164,34 @@ protected static void VerifyFlowGraphForTest<TSyntaxNode>(CSharpCompilation comp
                 stringBuilder.Builder.AppendLine($"    Statements ({statements.Length})");
                 foreach (var statement in statements)
                 {
+                    ValidateRoot(statement);
                     stringBuilder.Builder.AppendLine(OperationTreeVerifier.GetOperationTree(compilation, statement, initialIndent: 8));
                 }
 
                 if (block.Conditional.Condition != null)
                 {
                     Assert.True(map.TryGetValue(block.Conditional.Destination, out int index));
                     stringBuilder.Builder.AppendLine($"    Jump if {block.Conditional.JumpIfTrue} to Block[{index}]");
-                    stringBuilder.Builder.AppendLine(OperationTreeVerifier.GetOperationTree(compilation, block.Conditional.Condition, initialIndent: 8));
+
+                    IOperation condition = block.Conditional.Condition;
+                    ValidateRoot(condition);
+                    stringBuilder.Builder.AppendLine(OperationTreeVerifier.GetOperationTree(compilation, condition, initialIndent: 8));
+                }
+
+                void ValidateRoot(IOperation root)
+                {
+                    visitor.Visit(root);
+                    Assert.Null(root.Parent);
+                    Assert.Null(((Operation)root).SemanticModel);
+                    Assert.True(CanBeInControlFlowGraph(root), $"Unexpected node kind OperationKind.{root.Kind}");
```

Comment:
Thanks for the clarification. If `CanBeInControlFlowGraph` asserts, then we probably don't need to assert again here.